doc/modules/model_evaluation.rst

```diff
@@ -957,6 +921,55 @@ Similarly, labels not present in the data sample may be accounted for in macro-a
   ... # doctest: +ELLIPSIS
   0.166...
 
+.. _jaccard_score:
+
+Jaccard similarity coefficient score
+-------------------------------------
+
+The :func:`jaccard_score` function computes the average of `Jaccard similarity
+coefficients <https://en.wikipedia.org/wiki/Jaccard_index>`_, also called the
+Jaccard index, between pairs of label sets.
+
+The Jaccard similarity coefficient of the :math:`i`-th samples,
+with a ground truth label set :math:`y_i` and predicted label set
+:math:`\hat{y}_i`, is defined as
+
+.. math::
+
+    J(y_i, \hat{y}_i) = \frac{|y_i \cap \hat{y}_i|}{|y_i \cup \hat{y}_i|}.
+
+:func:`jaccard_score` works like :func:`precision_recall_fscore_support` as a
+naively set-wise measure applying natively to binary targets, and extended to
+apply to multilabel and multiclass through the use of `average` (see
+:ref:`above <average>`).
+
+In the multilabel case with binary label indicators: ::
+
+  >>> import numpy as np
+  >>> from sklearn.metrics import jaccard_score
+  >>> y_true = np.array([[0, 1, 1],
+  ...                    [1, 1, 0]])
+  >>> y_pred = np.array([[1, 1, 1],
+  ...                    [1, 0, 0]])
+  >>> jaccard_score(y_true[0], y_pred[0])  # doctest: +ELLIPSIS
+  0.6666...
+  >>> jaccard_score(y_true, y_pred, average='macro')  # doctest: +ELLIPSIS
+  0.6666...
+  >>> jaccard_score(y_true, y_pred, average='samples')  # doctest: +ELLIPSIS
+  0.5833...
+
+Multiclass problems are binarized and treated like the corresponding
+multilabel problem: ::
+
+  >>> y_pred = [0, 2, 1, 3]
+  >>> y_true = [0, 1, 2, 3]
+  >>> jaccard_score(y_true, y_pred, average='macro')
+  0.5
+  >>> jaccard_score(y_true, y_pred, average='micro')
+  0.33...
+  >>> jaccard_score(y_true, y_pred, average=None)
+  array([1., 0., 0., 1.])
```

Comment:
You're right. It's not common practice for evaluation, although Jaccard does have some nice properties that F1 (= Dice) does not, such as being the complement of a true distance metric.

And in whatever context, @bthirion had assumed `jaccard_similarity_score` would work on binary problems, since many people first know this index for set comparison.

It is, I admit, quite unusual for multiclass... and I'd be happy just to disallow it, and even throw away this PR and just allow multilabel.

Reply:
FWIW, I think P/R/F are pretty weird for multiclass in some ways too...

Reply:
> And in whatever context, @bthirion had assumed jaccard_similarity_score would work on binary problems, since many people first know this index for set comparison.

Seems that the wiki page we cite also demonstrates the binary case? (See https://en.wikipedia.org/wiki/Jaccard_index, Similarity of asymmetric binary attributes)

> It is, I admit, quite unusual for multiclass...

Agree. I'm happy to remove, or keep it for the sake of consistency.

> FWIW, I think P/R/F are pretty weird for multiclass in some ways too...

What do you mean? I guess it's reasonable for P/R/F to support multiclass?